docs/rules/no-async-describe.md

```diff
@@ -0,0 +1,47 @@
+# Disallow async functions passed to describe (no-async-describe)
+
+This rule disallows the use of an async function with `describe`. It usually indicates a copy/paste error or that you're trying to use `describe` for setup code, which should happen in `before` or `beforeEach`. Also, it can lead to [the contained `it` blocks not being picked up](https://github.com/mochajs/mocha/issues/2975).
+
+Example:
+
+```js
+describe(async function () {
+    // This work should happen in a beforeEach:
+    const theThing = await getTheThing();
+
+    it('should foo', function () {
+        // ...
+    });
+});
+```
+
+## Rule Details
+
+The rule supports "describe", "context" and "suite" suite function names and different valid suite name prefixes like "skip" or "only".
+
+The following patterns are considered problems, whether or not the function uses `await`:
+
+```js
+describe'something', async function () {
+    it('should work', function () {});
+});
+
+describe'something', async () => {
+    it('should work', function () {});
+});
+```
+
+If the `describe` function does not contain `await`, a fix of removing `async` will be suggested.
+
+This rule looks for every `describe.only`, `it.only`, `suite.only`, `test.only`, `context.only` and `specify.only` occurrences within the source code.
+Of course there are some edge-cases which can’t be detected by this rule, eg.:
+
+```js
+var describeOnly = describe.only;
+describeOnly.apply(describe);
```

Comment:
I’m not sure if this example makes sense. It is completely unrelated to `async`.

Reply:
I guess one edge case that we can’t detect might be something like this:

```js
async function mySuite() {
    it('my test', () => {});
}

describe('my suite', mySuite);
```

Reply:
D'oh, brain fart. I think I copy pasted from https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/test/rules/no-mocha-arrows.js#L15 and then my brain reset itself into a state where `describe` and `it` don't take a string argument.

Fixed in 7c9b6dc71664d8756fca64ddbbf0bf7657f727e7